### PR TITLE
Added try catch blocks where needed in impl layer

### PR DESCRIPTION
**Pull request by m-krajcovic**
Added try catch blocks for methods that use jdbc so they will throw InternalErrorException

### CODE REVIEW

File: perun-core/src/main/java/cz/metacentrum/perun/core/bl/ExtSourcesManagerBl.java [hunk 1]
```diff
@@ -251,5 +251,5 @@
 	 * @param extSource	External Source
 	 * @return			Map of attributes for external source
 	 */
-	Map<String, String> getAttributes(ExtSource extSource);
+	Map<String, String> getAttributes(ExtSource extSource) throws InternalErrorException;
```
Comment (stavamichal) [inline perun-core/src/main/java/cz/metacentrum/perun/core/bl/ExtSourcesManagerBl.java]: Missing information about InternalErrorException in javadoc.

File: perun-core/src/main/java/cz/metacentrum/perun/core/implApi/VosManagerImplApi.java [hunk 2]
```diff
@@ -218,7 +218,7 @@
 	 * @param sess
 	 * @param vo VO to delete all login reservations for
 	 */
-	public void deleteVoReservedLogins(PerunSession sess, Vo vo);
+	public void deleteVoReservedLogins(PerunSession sess, Vo vo) throws InternalErrorException;
```
Comment (stavamichal) [inline perun-core/src/main/java/cz/metacentrum/perun/core/implApi/VosManagerImplApi.java]: Missing information about InternalErrorException in javadoc.

File: perun-core/src/main/java/cz/metacentrum/perun/core/implApi/VosManagerImplApi.java [hunk 3]
```diff
@@ -206,7 +206,7 @@
 	 * @param appId from which application get reserved logins
 	 * @return list of pairs namespace and login
 	 */
-	public List<Pair<String, String>> getApplicationReservedLogins(Integer appId);
+	public List<Pair<String, String>> getApplicationReservedLogins(Integer appId) throws InternalErrorException;
```
Comment (stavamichal) [inline perun-core/src/main/java/cz/metacentrum/perun/core/implApi/VosManagerImplApi.java]: Missing information about InternalErrorException in javadoc.

File: perun-core/src/main/java/cz/metacentrum/perun/core/implApi/VosManagerImplApi.java [hunk 4]
```diff
@@ -197,7 +197,7 @@
 	 * @param vo
 	 * @return list of all vo applications ids
 	 */
-	public List<Integer> getVoApplicationIds(PerunSession sess, Vo vo);
+	public List<Integer> getVoApplicationIds(PerunSession sess, Vo vo) throws InternalErrorException;
```
Comment (stavamichal) [inline perun-core/src/main/java/cz/metacentrum/perun/core/implApi/VosManagerImplApi.java]: Missing information about InternalErrorException in javadoc.

File: perun-core/src/main/java/cz/metacentrum/perun/core/implApi/UsersManagerImplApi.java [hunk 5]
```diff
@@ -558,15 +558,15 @@
...
+	public List<Pair<String, String>> getUsersReservedLogins(User user) throws InternalErrorException;
 
 	/**
 	 * Delete all reserved logins for specific user
 	 * (pair is namespace and login)
 	 *
 	 * @param user for which get delete reserved logins
 	 */
-	public void deleteUsersReservedLogins(User user);
+	public void deleteUsersReservedLogins(User user) throws InternalErrorException;
```
Comment (stavamichal) [inline perun-core/src/main/java/cz/metacentrum/perun/core/implApi/UsersManagerImplApi.java]: Missing information about InternalErrorException in javadoc.

File: perun-core/src/main/java/cz/metacentrum/perun/core/implApi/UsersManagerImplApi.java [hunk 6]
```diff
@@ -558,15 +558,15 @@
 	 * @param user for which get reserved logins
 	 * @return list of pairs namespace and login
 	 */
-	public List<Pair<String, String>> getUsersReservedLogins(User user);
+	public List<Pair<String, String>> getUsersReservedLogins(User user) throws InternalErrorException;
```
Comment (stavamichal) [inline perun-core/src/main/java/cz/metacentrum/perun/core/implApi/UsersManagerImplApi.java]: Missing information about InternalErrorException in javadoc.

File: perun-core/src/main/java/cz/metacentrum/perun/core/implApi/GroupsManagerImplApi.java [hunk 7]
```diff
@@ -478,7 +478,7 @@
 	 * @param sess
 	 * @param group Group to delete all login reservations for
 	 */
-	public void deleteGroupReservedLogins(PerunSession sess, Group group);
+	public void deleteGroupReservedLogins(PerunSession sess, Group group) throws InternalErrorException;
```
Comment (stavamichal) [inline perun-core/src/main/java/cz/metacentrum/perun/core/implApi/GroupsManagerImplApi.java]: Missing information about InternalErrorException in javadoc.

File: perun-core/src/main/java/cz/metacentrum/perun/core/implApi/GroupsManagerImplApi.java [hunk 8]
```diff
@@ -466,7 +466,7 @@
 	 * @param appId from which application get reserved logins
 	 * @return list of pairs namespace and login
 	 */
-	public List<Pair<String, String>> getApplicationReservedLogins(Integer appId);
+	public List<Pair<String, String>> getApplicationReservedLogins(Integer appId) throws InternalErrorException;
```
Comment (stavamichal) [inline perun-core/src/main/java/cz/metacentrum/perun/core/implApi/GroupsManagerImplApi.java]: Missing information about InternalErrorException in javadoc.

File: perun-core/src/main/java/cz/metacentrum/perun/core/implApi/GroupsManagerImplApi.java [hunk 9]
```diff
@@ -457,7 +457,7 @@
 	 * @param group
 	 * @return list of all group applications ids
 	 */
-	public List<Integer> getGroupApplicationIds(PerunSession sess, Group group);
+	public List<Integer> getGroupApplicationIds(PerunSession sess, Group group) throws InternalErrorException;
```
Comment (stavamichal) [inline perun-core/src/main/java/cz/metacentrum/perun/core/implApi/GroupsManagerImplApi.java]: Missing information about InternalErrorException in javadoc.

File: perun-core/src/main/java/cz/metacentrum/perun/core/implApi/ExtSourcesManagerImplApi.java [hunk 10]
```diff
@@ -209,5 +209,5 @@
 	 * @param extSource	External Source
 	 * @return			Map of attributes for external source
 	 */
-	public Map<String,String> getAttributes(ExtSource extSource);
+	public Map<String,String> getAttributes(ExtSource extSource) throws InternalErrorException;
```
Comment (stavamichal) [inline perun-core/src/main/java/cz/metacentrum/perun/core/implApi/ExtSourcesManagerImplApi.java]: Missing information about InternalErrorException in javadoc.

**Comment by stavamichal** [comment]
After adding javadoc it seems to be ok.

**Comment by m-krajcovic** [comment]
updated with added javadocs

**Pull request merged**